lib/compiler-cranelift/src/compiler.rs

```diff
@@ -111,28 +110,36 @@ impl Compiler for CraneliftCompiler {
 
         let mut custom_sections = PrimaryMap::new();
 
-        #[cfg(all(target_arch = "x86_64", target_os = "linux"))]
-        let probestack_trampoline = CustomSection {
-            protection: CustomSectionProtection::ReadExecute,
-            // We create a jump to an absolute 64bits address
-            // with an indrect jump immediatly followed but the absolute address
-            // JMP [IP+0]   FF 25 00 00 00 00
-            // 64bits ADDR  00 00 00 00 00 00 00 00 preset to 0 until the relocation takes place
-            bytes: SectionBody::new_with_vec(vec![
-                0xff, 0x25, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-            ]),
-            relocations: vec![Relocation {
-                kind: RelocationKind::Abs8,
-                reloc_target: RelocationTarget::LibCall(LibCall::Probestack),
-                // 6 is the size of the jmp instruction. The relocated address must follow
-                offset: 6,
-                addend: 0,
-            }],
+        let probestack_trampoline_relocation_target = if target.triple().operating_system
+            == OperatingSystem::Linux
+            && matches!(
+                target.triple().architecture,
+                Architecture::X86_32(_) | Architecture::X86_64
+            ) {
```

Comment:
this code will not work on 32bits x86. the ff 25 00 00 00 00 translate to jmp [0x00000000].
Or the offset needs to be 2 instead of 6 in 32bits